frontend/src/actions/property.js

```diff
@@ -24,10 +24,41 @@ export function setPropertyField(fieldName, newValue) {
   };
 }
 
+export function setArrayValues(arrayName, item) {
+  let newArray = store.getState().propertyState.fields[arrayName];
+  let index = newArray.indexOf(item);
+  if (index > -1) {
+    newArray.splice(index, 1);
+  } else {
+    newArray.push(item);
+  }
+  store.dispatch(setPropertyField(arrayName, newArray));
+}
```

Comment:
this logic should be in the reducer and be neater, something like

field = [
 ...field,
  newItem 
]